### ISSUE TITLE
vue/prop-name-casing false positive reaction

### ISSUE THREAD

**Issue by yekver** [vuejs/eslint-plugin-vue#932]
**Tell us about your environment**	
* **ESLint version: 6.0.1**	
* **eslint-plugin-vue version: 5.2.3**	
* **Node version: 12.6.0**	

**What did you do?**	

```vue
export default {
  props: {
    $css: { type: Function, default: () => '' },
  }
}
```

**What did you expect to happen?**
No warning should be emitted

**What actually happened?**
`Prop "$css" is not in camelCase.eslint(vue/prop-name-casing)`

**Comment by sirlancelot** [comment]
The same is happening for props prefixed with `_` as well:

![image](https://user-images.githubusercontent.com/71916/62729940-d8ea9100-b9d3-11e9-85f9-ab83a64f09a9.png)

Even worse, the linter is automatically "fixing" this which actually breaks code that would depend on the underscore being there. I'm using the underscore as a way to represent "internal" props which should never be accessed from templates.

**Comment by sirlancelot** [comment]
Related: #862
